pennylane/utils.py

```diff
@@ -56,32 +60,30 @@ def _unflatten(flat, model):
     See also :func:`_flatten`.
 
     Args:
-        flat (array): 1D array of items
-        model (array, Iterable, Number): model nested structure
+        flat (array[Any]): 1D array of items
+        model (Iterable, Any): model nested structure
 
     Raises:
         TypeError: if ``model`` contains an object of unsupported type
 
     Returns:
-        Union[array, list, Any], array: first elements of flat arranged into the nested
+        Union[array, list, Any], array: first elements of `flat` arranged into the nested
         structure of model, unused elements of flat
     """
-    if isinstance(model, (numbers.Number, VariableRef, str)):
-        return flat[0], flat[1:]
-
```

Comment:
I don't remember why this was restricted to just these types.

Reply:
Hmm, that's odd, were these lines not covered by the test suite before?

Reply:
I disabled the test.